sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -299,6 +299,27 @@ private[sql] trait SQLTestUtilsBase
     }
   }
 
+  /**
+   * Drops cache `cacheName` after calling `f`.
+   */
+  protected def withCache(cacheNames: String*)(f: => Unit): Unit = {
+    try f finally {
+      cacheNames.foreach(uncacheTable)
+    }
+  }
```

Comment:
Could you update like the following? Like the other `withXXX` functions, `withCache` should clean up multiple cache names in the `finally` clause.
```scala
   protected def withCache(cacheNames: String*)(f: => Unit): Unit = {
     try f finally {
-      cacheNames.foreach(uncacheTable)
+      cacheNames.foreach { cacheName =>
+        try uncacheTable(cacheName) catch {
+          case _: AnalysisException =>
+        }
+      }
     }
   }
```

Reply:
The above is required when we use `withCache("a", "b", "c")`.